Owlbot v2

UmiLovesU2 opened the pull request:
Changes implemented as of 21/03/2024 are as follows.

1. New naming convention reflecting scope change on the bot's functions
2. Event-based automated message that welcomes new members of the server
3. Basis for scheduled event-based warning system for incoming events, with incorporated exception handling, this fixes the bot from breaking in case there's no events when it pulls them from the list.

UmiLovesU2 committed:
Added automated messaging process on member join, removed member ping before mission event as the automation will now be implemented via discord events.

UmiLovesU2 committed:
Barebones Event Functionality, specialized event warning to users using embeds on the pipeline.

Exception handling in case there's no events on the server when the bot wakes up.

UmiLovesU2 committed:
Basic event functionality implemented, still not finished.

Eventualities such as edited events are still not contemplated, and are TODO.

Event deletions, manual event starts and edits to discord events are not detected.

background task that checks if the event has started unconsistently throws Exception due to ts.ip being detected as a url.

on_message event is a test method that will be deleted later.

UmiLovesU2 commented:
New commit implements warning system based on discord events, however issues are:

- Eventualities such as edited events are still not contemplated, and are TODO.
- Event deletions, manual event starts and edits to discord events are not detected.
- background task that checks if the event has started unconsistently throws Exception due to ts.ip being detected as a url.
- on_message event is a test method that will be deleted later.

enrico-ghidoni commented:
Thanks a lot for the contribution Umi! Love seeing some love bombing on the OWLBOT.

As discussed on TeamSpeak I'd rather keep the event notification trigger based on cron instead of Discord events. This is for a couple reasons:

- There's no guarantee events are actually going to be on Discord, so triggering join reminders based on that could prove unreliable sometimes. Mind you, automating stub event creations is indeed on the todo-list; ideally Discord community events and Forum's calendar should be automated and I'm absolutely on board with OWLBOT doing both things. But I'd like to see a separate module of it dealing with their creation and once that is done we can think of community event-based notifications. It's also very noticeable how handling community events without structuring the thing is [verbose and requires a lot of checks](https://github.com/CntoDev/discord-owlbot/pull/1/files#diff-5ffcc5c595bdb5127f82e03dbcdb1eeb75a2633f6f44b7f640f9d7523bc0f264R71).
- Discord community events are set with a starting time according to our Rules & Regulations, which is at 20:00 CET / CEST. Reminders have proven effective because they are some time before we are supposed to join the server thus giving people enough time to perform quick game / repo updates and get to their PCs.
- It's easier, at least for now.

What this means is that OWLBOT should have a separate entrypoint to allow bash commands to send notifications. Something like OWLBOT v1 should work. An example could be the following:
`python owlbot.py op-notification`, the `op-notification` invocation parameter can be caught by the [argparse](https://docs.python.org/3/library/argparse.html) library.

So, a few more things before we merge this into stable release if you don't mind.

1. Clean up unnecessary code and testing functions
2. Remove the OP is about to start notification from the event loop and make a separate function to send a channel message whenever the script is called with the appropriate argument.

UmiLovesU2 committed:
Code refactor removing all event-related testing has been removed until further expansion on the OWLBOT's functions.

role and channel IDs related to the on_member_join event embed message are now environment variables.

TODO: work on a method that creates embeds to follow DRY principle.

enrico-ghidoni commented:
Superseded by https://github.com/CntoDev/discord-owlbot/pull/3
